README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# About

[![Travis-CI Build
Status](https://travis-ci.org/jjchern/sahie.svg?branch=master)](https://travis-ci.org/jjchern/sahie)

See the [`sahieAPI` package](https://github.com/jjchern/sahieAPI) for
details.

# Installation

``` r
# install.packages("remotes")
remotes::install_github("jjchern/sahie")
```


NEWS.md

# sahie 0.0.2

* Deleted `county_a5064`.
* Added `county_yr????_ag????`.
* Added `county_yr????_ag????_pv??????`.

# sahie 0.0.1

* Added a `NEWS.md` file to track changes to the package.
* Added `county_a5064`.


R/data.R
#' @title County-Level Number of Uninsured Eestimates
#' @description Use \code{sahie_county}.
"sahie_county"

# By age ------------------------------------------------------------------

#' County-Level Estimates of Health Insurance Coverage: Age 00-64
#'
#' Available years: 2008-2016. Estimates are specific to those age under
#' 65, are of any race, any sex, and for all income levels.
"county_yr0816_ag0064"

#' County-Level Estimates of Health Insurance Coverage: Age 18-64
#'
#' Available years: 2008-2016. Estimates are specific to those age 18-64,
#' are of any race, any sex, and for all income levels.
"county_yr0816_ag1864"

#' County-Level Estimates of Health Insurance Coverage: Age 40-64
#'
#' Available years: 2008-2016. Estimates are specific to those age 40-64,
#' are of any race, any sex, and for all income levels.
"county_yr0816_ag4064"

#' County-Level Estimates of Health Insurance Coverage: Age 50-64
#'
#' Available years: 2010-2016. Estimates are specific to those age 50-64,
#' are of any race, any sex, and for all income levels.
"county_yr1016_ag5064"

#' County-Level Estimates of Health Insurance Coverage: Age 00-18
#'
#' Available years: 2008-2016. Estimates are specific to those age 00-18,
#' are of any race, any sex, and for all income levels.
"county_yr0816_ag0018"

#' County-Level Estimates of Health Insurance Coverage: Age 21-64
#'
#' Available years: 2014-2016. Estimates are specific to those age 21-64,
#' are of any race, any sex, and for all income levels.
"county_yr1416_ag2164"

# By age and income -------------------------------------------------------

#' County-Level Estimates of Health Insurance Coverage: Age 00-64 and
#' Below 200\% of poverty
#'
#' Available years: 2008-2016. Estimates are specific to those age under
#' 65, are of any race, any sex, and with income at or below 200\% of poverty.
"county_yr0816_ag0064_pv000200"

#' County-Level Estimates of Health Insurance Coverage: Age 18-64 and
#' Below 200\% of poverty
#'
#' Available years: 2008-2016. Estimates are specific to those age 18-64,
#' are of any race, any sex, and with income at or below 200\% of poverty.
"county_yr0816_ag1864_pv000200"

#' County-Level Estimates of Health Insurance Coverage: Age 40-64 and
#' Below 200\% of poverty
#'
#' Available years: 2008-2016. Estimates are specific to those age 40-64,
#' are of any race, any sex, and with income at or below 200\% of poverty.
"county_yr0816_ag4064_pv000200"

#' County-Level Estimates of Health Insurance Coverage: Age 50-64 and
#' Below 200\% of poverty
#'
#' Available years: 2010-2016. Estimates are specific to those age 50-64,
#' are of any race, any sex, and with income at or below 200\% of poverty.
"county_yr1016_ag5064_pv000200"

#' County-Level Estimates of Health Insurance Coverage: Age 00-18 and
#' Below 200\% of poverty
#'
#' Available years: 2008-2016. Estimates are specific to those age 00-18,
#' are of any race, any sex, and with income at or below 200\% of poverty.
"county_yr0816_ag0018_pv000200"

#' County-Level Estimates of Health Insurance Coverage: Age 21-64 and
#' Below 200\% of poverty
#'
#' Available years: 2014-2016. Estimates are specific to those age 21-64,
#' are of any race, any sex, and with income at or below 200\% of poverty.
"county_yr1416_ag2164_pv000200"


data-raw/prep.R

library(tidyverse)
library(tidylog)

# Download raw data -------------------------------------------------------
# https://www.census.gov/data/datasets/time-series/demo/sahie/estimates-acs.html

glue::glue("https://www2.census.gov/programs-surveys/sahie/datasets/time-series/estimates-acs/sahie-{2008:2016}-csv.zip") %>% print() -> zip_urls

glue::glue("data-raw/sahie_{2008:2016}.csv") %>% print() -> fils

map2(.x = fils,
     .y = zip_urls,
     ~ if (!file.exists(.x)) {
            temp = tempfile(fileext = ".zip")
            download.file(.y, temp)
            unzip(temp, exdir = "data-raw", junkpaths = TRUE)
     })

# 2013 doesn't have 21 to 64 years [5]
# usethis::use_data(sahie_county, overwrite = TRUE)

#' From 2016
#' geocat: 2 Geography category
#' 40 - State geographic identifier
#' 50 - County geographic identifier
#'
#' agecat: 1 Age category
#' 0 - Under 65 years
#' 1 - 18 to 64 years
#' 2 - 40 to 64 years
#' 3 - 50 to 64 years
#' 4 - Under 19 years
#' 5 - 21 to 64 years
#'
#' racecat: 1 Race category
#' 0 - All races
#' Only state estimates have racecat=1
#' 1 - White alone
#' 2 - Black alone
#' 3 - Hispanic (any race)
#'
#' sexcat: 1 Sex category
#' 0 - Both sexes
#' 1 - Male
#' 2 - Female
#'
#' iprcat: 1 Income category
#' 0 - All income levels
#' 1 - At or below 200% of poverty
#' 2 - At or below 250% of poverty
#' 3 - At or below 138% of poverty
#' 4 - At or below 400% of poverty
#' 5 - Between 138% - 400%  of poverty
#'
#' nipr        : Number in demographic group for <income category>
#' nipr_moe    : MOE for nipr
#' nui         : Number uninsured
#' nui_moe     : MOE for nui
#' niu         : Number insured
#' nic_moe     : MOE for niu
#' pctui       : Percent uninsured in demographic group for <income category>
#' pctui_moe   : MOE for pctui
#' pctic       : Percent insured in demographic group for <income category>
#' pctic_moe   : MOE for pctic
#' pctelig     : Percent uninsured in demographic group for all income levels
#' pctelig_moe : MOE for pctelig
#' pctliic     : Percent insured in demographic group for all income levels
#' pctliic_moe : MOE for pctliic
#' state_name  : State Name
#' county_name : County Name


data-raw/prep-county-a5064.R

# Prepareing county-level uninsured rate for age 50-64.
# All races, both sexes, and all income categories
# Available years: 2008-2016

prep = . %>%
        select(-version, -X26) %>%
        docxtractr::mcga() %>%
        mutate(fips = paste0(statefips, countyfips)) %>%
        # County-level
        filter(geocat == 50) %>%
        # age 50-64 [3]; 21 to 64 years [5]; 18 to 64 years [1]
        filter(agecat == 1) %>%
        # all races
        filter(racecat == 0) %>%
        # both sexes
        filter(sexcat == 0) %>%
        # all income level [0]; At or below 200% of poverty [1]
        # also no need for contains("elig") and contains("liic")
        filter(iprcat == 0) %>%
        select(-contains("elig"), -contains("liic")) %>%
        select(-geocat, -agecat, -racecat, -sexcat, -iprcat) %>%
        select(-statefips, -countyfips) %>%
        select(fips, year, state = state_name, county = county_name, everything())

bind_rows(
        read_csv("data-raw/sahie_2016.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2015.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2014.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2013.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2012.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2011.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2010.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2009.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2008.csv", skip = 79) %>% prep() %>% print()
) %>%
        arrange(fips, year) %>%
        print() -> county_a5064

use_data(county_a5064)


data-raw/prep-county-by-ag-by-pv.R

# Prepareing county-level estimates for HI coverage for
# specific age groups and income levels.
# All races, both sexes
# Available years: 2008-2016

prep = . %>%
        select(-version, -X26) %>%
        docxtractr::mcga() %>%
        mutate(fips = paste0(statefips, countyfips)) %>%
        # County-level
        filter(geocat == 50) %>% select(-geocat) %>%
        # all races
        filter(racecat == 0) %>% select(-racecat) %>%
        # both sexes
        filter(sexcat == 0) %>% select(-sexcat) %>%
        # no need for contains("elig") and contains("liic")
        select(-contains("elig"), -contains("liic")) %>%
        select(-statefips, -countyfips) %>%
        select(fips, year, state = state_name, county = county_name, everything())

bind_rows(
        read_csv("data-raw/sahie_2016.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2015.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2014.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2013.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2012.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2011.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2010.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2009.csv", skip = 79) %>% prep() %>% print(),
        read_csv("data-raw/sahie_2008.csv", skip = 79) %>% prep() %>% print()
) %>%
        arrange(fips, year) %>%
        print() -> county_raw

# Save county-leve files by available age categories ----------------------

county_raw %>% filter(iprcat == 0) %>% count(year, agecat) %>% spread(agecat, n)
#' agecat: Age category
#' 0 - Under 65 years
#' 1 - 18 to 64 years
#' 2 - 40 to 64 years
#' 3 - 50 to 64 years
#' 4 - Under 19 years
#' 5 - 21 to 64 years

county_raw %>% filter(iprcat == 0, agecat == 0) %>% print() -> county_yr0816_ag0064
county_raw %>% filter(iprcat == 0, agecat == 1) %>% print() -> county_yr0816_ag1864
county_raw %>% filter(iprcat == 0, agecat == 2) %>% print() -> county_yr0816_ag4064
county_raw %>% filter(iprcat == 0, agecat == 3) %>% print() -> county_yr1016_ag5064
county_raw %>% filter(iprcat == 0, agecat == 4) %>% print() -> county_yr0816_ag0018
county_raw %>% filter(iprcat == 0, agecat == 5) %>% print() -> county_yr1416_ag2164

use_data(county_yr0816_ag0064,
         county_yr0816_ag1864,
         county_yr0816_ag4064,
         county_yr1016_ag5064,
         county_yr0816_ag0018,
         county_yr1416_ag2164, overwrite = TRUE)

# Save county-leve files by available age and income categories ---------------

county_raw %>%
        count(year, iprcat, agecat) %>%
        spread(agecat, n) %>%
        arrange(iprcat, year) %>%
        print(n = 50)
#' iprcat: 1 Income category
#' 0 - All income levels
#' 1 - At or below 200% of poverty
#' 2 - At or below 250% of poverty
#' 3 - At or below 138% of poverty
#' 4 - At or below 400% of poverty
#' 5 - Between 138% - 400%  of poverty

county_raw %>% filter(iprcat == 1, agecat == 0) %>% print() -> county_yr0816_ag0064_pv000200
county_raw %>% filter(iprcat == 1, agecat == 1) %>% print() -> county_yr0816_ag1864_pv000200
county_raw %>% filter(iprcat == 1, agecat == 2) %>% print() -> county_yr0816_ag4064_pv000200
county_raw %>% filter(iprcat == 1, agecat == 3) %>% print() -> county_yr1016_ag5064_pv000200
county_raw %>% filter(iprcat == 1, agecat == 4) %>% print() -> county_yr0816_ag0018_pv000200
county_raw %>% filter(iprcat == 1, agecat == 5) %>% print() -> county_yr1416_ag2164_pv000200

use_data(county_yr0816_ag0064_pv000200,
         county_yr0816_ag1864_pv000200,
         county_yr0816_ag4064_pv000200,
         county_yr1016_ag5064_pv000200,
         county_yr0816_ag0018_pv000200,
         county_yr1416_ag2164_pv000200, overwrite = TRUE)
